spec.md

```diff
@@ -0,0 +1,238 @@
+
+This includes individual artifacts that use [OCI manifest][image-manifest] including [OCI Image][image-spec], [Helm][helm] and [Singularity][singularity].
+
+### Future Scope
+
+Future versions of artifacts will support collections of artifacts using [OCI Index][image-index].
+
+To support artifacts that represent a collection of other artifacts, a means to identify an index is a type of artifact will be required.
+
+## Defining OCI Artifact Types
+
+As registries become content addressable distribution points, tools that pull artifacts must know if they can operate on the artifact. Artifact types are equivalent to file extensions. When users open files, the host operating system typically launches the appropriate program. When users open a file, from within a program, the open dialog filters to the supported types. When search or security software scan the contents of a storage solution, the software must to know how to process the different types of content. When users view the contents of a storage solution, they see the textual and visual indications of the type. OCI Artifacts provides these core capabilities to [OCI distribution spec][distribution-spec] based registries.
+
+Authoring an OCI Artifacts involves the following steps:
+
+- Define a unique type
+- Define the format for other tools to operate upon the type
+- Define human elements, such as an icon and localized string to be displayed to users
+- Optionally, publish as a [well-known type][def-well-known-types] for registries to consume
+
+Defining a unique type, string and logo; artifacts can be represented as the following:
+
+|Icon|Artifact|`config.mediaType`|
+|-|-|-|
+|<img src="https://github.com/opencontainers/artwork/blob/master/oci/icon/color/oci-icon-color.png?raw=true" width=30x>|[OCI Image][image-spec]|`application/vnd.oci.image.config.v1+json`|
+|<img src="https://github.com/helm/helm-www/blob/master/themes/helm/static/img/apple-touch-icon.png?raw=true" width=30x>|[Helm Chart](https://helm.sh)|`application/vnd.cncf.helm.chart.config.v1+json`|
+|<img src="https://github.com/sylabs/singularity/blob/master/docs/logos/singularity_v3.png?raw=true" width=30x>|[Singularity][singularity], by [Sylabs][sylabs]|`application/vnd.sylabs.sif.config.v1+json`|
+
+## Defining a Unique Artifact Type
+
+Defining the unique type involves uniqueness for computer processing, and uniqueness for humans.
+
+For computer processing, artifacts are defined by setting `manifest.config.mediaType` to a globally unique value.
+
+> **Note:** The `config.mediaType` of `application/vnd.oci.image.config.v1+json` is reserved for artifacts intended to be instanced by [docker][docker] and [containerd][containerd].  
+*Each artifact MUST have its own unique type.*
+
+ The following `mediaType` format is used to differentiate the type of artifact:
+
+`application/vnd.`[org|company]`.`[objectType]`.`[optionalSubType]`.config.`[version]`+json`
```

Comment:
Do we need this?  I'd rather punt to [RFC 6838][1] to define media type syntax and registration.  I see no reason folks couldn't push image/png, text/plain, etc. into an artifact store.

[1]: https://tools.ietf.org/html/rfc6838

Reply:
Yes, you can push your family photos if you’d like. We’re just saying there should be a PictureLibrary artifact type defined. 
As a registry operator, we want to understand what are the contents of the registry. Enforcing a definition type enables registries to provide information to registry products, like security scanning that need to know the type to process it. Without known types, scanners will just see things it has to guess at, minimizing the value. 
A registry operator may choose to leave their manifest validation open. The spec provides the standard, by which operators can choose from.

Reply:
> We’re just saying there should be a PictureLibrary artifact type defined.

But that doesn't need your media-type string restrictions.  You could just choose to reject image/png uploads until someone had registered the type with you and told you how to scan PNGs for security issues, identify non-PNG content claiming to be image/png, etc.  I don't see how requiring an `application/vnd.` prefix, etc., helps with any of that.

Reply:
Ahh, ok. You're asking why we start with `application/vnd`, vs something like `image/png`
This unique naming section has been the source of some good questions and curiosity. Let me read through RFC 6838, but wow, big. We'll need something shorter for guidance.

Reply:
RFC 6838 looks promising. I'm leaving this one as unresolved to tend to the rest of the feedback. But, I'll come back and address this across the spec shortly.

Reply:
I also wonder why the `.config.` fragment is needed... Isn't it the artifact type? The artifact type is not a configuration.

Requiring JSON (`+json`) is IMHO not needed, not even for configurations. What about using YAML, INI,... for configuration?

Reply:
> I also wonder why the `.config.` fragment is needed... Isn't it the artifact type? The artifact type is not a configuration.
> 
Having an actual config object is an optional element. Using the `manifest.config.mediaType` was the best thing we could use, that didn't break everyone. The more we discussed it, including having an optional config object, the more folks got comfortable with it. Once you have several artifact types, the picture starts to get more crisp.

> Requiring JSON (`+json`) is IMHO not needed, not even for configurations. What about using YAML, INI,... for configuration?

We did agree that the config object could be whatever the artifact type defines. Including yaml, ini, or any type of text format. While it could be binary, we wanted the config object to be easily parseable by registries, as they historically look into the config object for some metadata the registries index.

Reply:
I've added info on when to use `.json`, `.yaml` or not provide the extension for null config objects.